packages/frontend/src/components/DashboardFilter/index.tsx

```diff
@@ -34,65 +31,75 @@ const DashboardFilter: FC<Props> = ({ isEditMode }) => {
                     disabled={!hasChartTiles}
-                    canEscapeKeyClose={isSubmenuOpen ? false : true}
-                    interactionKind={isSubmenuOpen ? 'click-target' : 'click'}
-                    placement="bottom-start"
-                    onOpened={() => setIsOpen(true)}
-                    onClose={handleClose}
-                    content={
+                    onClose={() => setIsFilterPopoverOpen(false)}
+                    transitionProps={{
+                        transition: 'pop',
+                    }}
+                    withArrow
+                    shadow="md"
+                    offset={-1}
+                >
+                    <Popover.Target>
+                        <Tooltip
+                            disabled={isFilterPopoverOpen || isEditMode}
+                            position="bottom"
+                            openDelay={500}
+                            label={
+                                <Text fz="xs">
+                                    Only filters added in <b>'edit'</b> mode
+                                    will be saved
+                                </Text>
+                            }
+                        >
+                            <Button
+                                size="xs"
+                                variant="default"
+                                leftIcon={
+                                    <MantineIcon
+                                        color="blue"
+                                        icon={IconFilter}
+                                    />
+                                }
+                                disabled={!hasChartTiles}
+                                onClick={() =>
+                                    setIsFilterPopoverOpen(
+                                        (prevIsOpen) => !prevIsOpen,
+                                    )
+                                }
+                            >
+                                Add filter
+                            </Button>
+                        </Tooltip>
+                    </Popover.Target>
+
+                    <Popover.Dropdown ml={5} p={0}>
                         <FilterSearch
                             isEditMode={isEditMode}
                             fields={allFilterableFields || []}
                             popoverProps={{
-                                onOpened: () => setIsSubmenuOpen(true),
-                                onOpening: () => setIsSubmenuOpen(true),
-                                onClose: () => setIsSubmenuOpen(false),
-                                onClosing: () => setIsSubmenuOpen(false),
+                                usePortal: false,
                             }}
-                            onClose={handleClose}
-                            onSelectField={handleClose}
+                            onClose={() => setIsFilterPopoverOpen(false)}
+                            onSelectField={() => {}}
```

Comment:
Future work?

Reply:
we might have to address this on the next piece of work! For now, it should be no-op 👍